src/tests/multihost/alltests/conftest.py

```diff
@@ -1356,6 +1356,66 @@ def remove_host_entry():
             (_, _) = ldap_inst.del_dn(dn)
     request.addfinalizer(remove_host_entry)
 
+
+@pytest.fixture(scope='class')
+def ns_account_lock(session_multihost, request):
+    """ Backup and restore sssd.conf
+        Add managed role
+        Add filter role
+    """
+    tools = sssdTools(session_multihost.client[0])
+    domain_name = tools.get_domain_section_name()
+    client = sssdTools(session_multihost.client[0])
+    domain_params = {'cache_credentials': 'true',
+                     'enumerate': 'true',
+                     'access_provider': 'ldap',
+                     'ldap_access_order': 'expire',
+                     'ldap_account_expire_policy': '389DS',
+                     'ldap_ns_account_lock': 'nsAccountlock'}
+    client.sssd_conf(f'domain/{domain_name}', domain_params)
+    domain_params = {'reconnection_retries': '3'}
+    client.sssd_conf('pam', domain_params)
+    domain_params = {'filter_groups': 'root',
+                     'filter_users': 'root',
+                     'reconnection_retries': '3',
+                     'debug_level': '9'}
+    client.sssd_conf('nss', domain_params)
+    session_multihost.client[0].service_sssd('restart')
+    # Add managed role
+    master_e = session_multihost.master[0].ip
+    ldap_uri = f'ldap://{master_e}'
+    ds_rootdn = 'cn=Directory Manager'
+    ds_rootpw = 'Secret123'
+    ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+    user_info = {'cn': 'managed'.encode('utf-8'),
+                 'objectClass': [b'top', b'LdapSubEntry',
+                                 b'nsRoleDefinition',
+                                 b'nsSimpleRoleDefinition',
+                                 b'nsManagedRoleDefinition']}
+    user_dn = 'cn=managed,ou=People,dc=example,dc=test'
+    (_, _) = ldap_inst.add_entry(user_info, user_dn)
+    # Add filter role
+    ldap_uri = f'ldap://{master_e}'
+    ds_rootdn = 'cn=Directory Manager'
+    ds_rootpw = 'Secret123'
+    ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+    user_info = {'cn': 'filtered'.encode('utf-8'),
+                 'objectClass': [b'top', b'LdapSubEntry',
+                                 b'nsRoleDefinition',
+                                 b'nsComplexRoleDefinition',
+                                 b'nsFilteredRoleDefinition'],
+                 'nsRoleFilter': 'o=filtered'.encode('utf-8'),
+                 'Description': 'filtered role'.encode('utf-8')}
+    user_dn = 'cn=filtered,ou=People,dc=example,dc=test'
+    (_, _) = ldap_inst.add_entry(user_info, user_dn)
+
+    def restoresssdconf():
+        """ Restore sssd.conf """
+        restore = 'cp -f %s.orig %s' % (SSSD_DEFAULT_CONF, SSSD_DEFAULT_CONF)
+        session_multihost.client[0].run_command(restore)
+    request.addfinalizer(restoresssdconf)
```

Comment:
Should include cleanup of the roles

Reply:
for i in ['cn=managed,ou=people,dc=example,dc=test',
                  'cn=filtered,ou=people,dc=example,dc=test',
                  'cn=nested,ou=People,dc=example,dc=test']:
            ldap_inst.del_dn(i)

Already done

Reply:
Any way now i am moving this part to fixture